src/backend/backend.ts

```diff
@@ -106,8 +127,34 @@ const updateComponentTree = (roots: Array<any>) => {
   previousCount = count;
 };
 
-const updateRouterTree = (routes: Array<MainRoute>) => {
-  messageBuffer.enqueue(MessageFactory.routerTree(routes));
+const updateLazyLoadedNgModules = (routers) => {
+
+  routers.forEach(router => {
+    parseModulesFromRouter(router, parsedModulesData);
+  });
+
+  sendNgModulesMessage();
+};
+
+const updateRouterTree = () => {
+  const routers: Array<any> = routerTree();
+  const parsedRoutes = routers.map(parseRoutes);
+
+  let routesChanged = !previousRoutes ? true : false;
+
+  if (previousRoutes) {
+    const changes = compare(previousRoutes, parsedRoutes);
```

Comment:
Use `compare` to check for changes in the router and only send data to the frontend if there are any.